lib/ecto/pools/sojourn_broker/worker.ex

```diff
@@ -41,6 +41,9 @@ defmodule Ecto.Pools.SojournBroker.Worker do
     backoff_threshold = div(max_backoff, 3)
     lazy = Keyword.get(worker_opts, :lazy, true)
 
+    # Seed random number generator for backoff
+    :random.seed(:erlang.now)
```

Comment:
`:erlang.now/0` will generate a warning in 18.0+ so we shouldn't introduce it.

This 18.0 docs for `:random` suggest:
```erlang
random:seed(erlang:phash2([node()]), erlang:monotonic_time(), erlang:unique_integer())
```

I guess we could use the has of a reference for a unique enough number:
```elixir
{_, Sec, Micro} = :os.timestamp()
{Sec, Micro, :erlang.phash2([make_ref()])}
```